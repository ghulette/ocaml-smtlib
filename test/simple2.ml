open Easy_smt.Smtlib
open Smtlib_test

let test1 solv =
  let int_sort = Sort (Id "Int") in
  declare_const solv (Id "x") int_sort;
  declare_const solv (Id "y") int_sort;
  let x = const "x" in
  let y = const "y" in
  assert_ solv (equals x (int_to_term 7));
  assert_ solv (equals y (add x (int_to_term 1)))

let () =
  run_test test1
