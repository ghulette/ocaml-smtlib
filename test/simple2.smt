(declare-fun int_of_bool (Bool) Int)
(assert (= (int_of_bool false) 0))
(assert (= (int_of_bool true) 1))
(declare-fun bool_of_int (Int) Bool)
(assert (forall ((x Int)) (=> (>= 0 x) (bool_of_int x) true)))
(assert (forall ((x Int)) (=> (< 0 x) (bool_of_int x) false)))
(check-sat)
(get-model)
