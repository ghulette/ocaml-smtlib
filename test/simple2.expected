sat
int_of_bool(x!0 : Bool) : Int := (ite (= x!0 true) 1 0)
